### PR TITLE
feature/MASSDE-181 : Disable Carthage update if no Cartfile is present

### PR DESCRIPTION
**Pull request by shoebox**

### CODE REVIEW

File: plugin/src/main/groovy/org/openbakery/carthage/CarthageUpdateTask.groovy [hunk 1]
```diff
@@ -1,71 +1,109 @@
...
+			project.rootProject.file("Carthage/Build/" + getCarthagePlatformName())
+		}
+	}
+
+	@TaskAction
+	void update() {
+		if (hasCartFile()) {
+			logger.info('Update Carthage for platform ' + carthagePlatformName)
+
+			def output = services.get(StyledTextOutputFactory).create(CarthageUpdateTask)
```
Comment (elsassph) [inline plugin/src/main/groovy/org/openbakery/carthage/CarthageUpdateTask.groovy]: So no skip if the `/Carthage` folder exists?

File: plugin/src/main/groovy/org/openbakery/carthage/CarthageUpdateTask.groovy [hunk 2]
```diff
@@ -1,71 +1,109 @@
...
-
-		def carthagePlatform = getCarthagePlatform()
-		logger.info('Update Carthage for platform ' + carthagePlatform)
-
-		File carthageDirectory = new File(project.projectDir, "Carthage")
-		File platformDirectory = carthagePlatform != 'all' ? new File(new File(carthageDirectory, 'Build'), carthagePlatform) : null
-		if (carthageDirectory.exists() && (platformDirectory != null && platformDirectory.exists())) {
-			logger.info('Skip Carthage update')
-			return
+	@InputFile
```
Comment (elsassph) [inline plugin/src/main/groovy/org/openbakery/carthage/CarthageUpdateTask.groovy]: Input should also mention the target platform

File: plugin/src/main/groovy/org/openbakery/carthage/CarthageUpdateTask.groovy (hunk 1 above)
Comment (shoebox) [reply plugin/src/main/groovy/org/openbakery/carthage/CarthageUpdateTask.groovy]: Gradle will automatically skip the task if the current status of the inputs (`cartfile` and `cartfile.resolved`) and the output folder are still in sync with the task cache.
cf https://docs.gradle.org/current/userguide/more_about_tasks.html for more details

**Comment by renep** [comment]
What the status of this pull request. Should I merge it and see if my build pipeline works?
The pipeline builds the plugin and then the example projects with the new plugin version. And if this works and all tests are green, then a new develop version is automatically released on the openbakery.org repo.

**Comment by shoebox** [comment]
@renep Please proceed, I have myself been running some tests and everything seems to be running as expected (with or without Carthage in the project).
I too tried to setup some functional unit tests with gradle testkit, but the plugin resolution is failing, so the tests too.
So until we find a way to make those functional unit tests working, your testbed is probably one of the best way to ensure everything is functional.

**Comment by renep** [comment]
Merge is a new develop version is in the openbakery repo: (see https://github.com/openbakery/gradle-xcodePlugin/tree/develop#current-develop-version)

**Pull request merged**